### DISCUSSION TITLE
node ace build stuck in server

### DISCUSSION THREAD

**Discussion by mgarcialeniolabs** [adonisjs/core#4752] [Help]
Hey guys! Been using adonisjs for a while now... but recently I've started experiencing, on one of my apps, this issue where the compiler get stuck.

![Screenshot 2024-10-09 at 3 53 45 PM](https://github.com/user-attachments/assets/0b4ac54b-cef4-4e60-966a-fa8f3ea2ad1e)

This app was working correctly and now it hangs in that command.

**Comment by thetutlage** [comment]
Can you run `tsc` directly and see how that goes? It seems like that the TypeScript compiler is taking ages to resolve some typings and that could be because of many reasons.

**Reply by mgarcialeniolabs** [reply]
`./node_modules/.bin/tsc` runs okay in a few seconds
`./node_modules/.bin/tsc --noEmit` runs okay in a few seconds
`./node_modules/.bin/tsc --diagnostics` see below

```
Files:              917
Lines:           175189
Identifiers:     170238
Symbols:         166283
Types:            19273
Instantiations:   53943
Memory used:    204920K
I/O read:         0.06s
I/O write:        0.02s
Parse time:       2.65s
Bind time:        0.97s
Check time:       2.02s
Emit time:        0.22s
Total time:       5.86s
```

**Comment by mgarcialeniolabs** [comment]
Solved. There was an issue with the VM itself.
